saleor/graphql/discount/mutations/promotion_create.py

```diff
@@ -0,0 +1,223 @@
+        description = "Creates a new promotion." + ADDED_IN_315 + PREVIEW_FEATURE
+        model = models.Promotion
+        object_type = Promotion
+        permissions = (DiscountPermissions.MANAGE_DISCOUNTS,)
+        error_type_class = PromotionCreateError
+        doc_category = DOC_CATEGORY_DISCOUNTS
+
+    @classmethod
+    def clean_input(
+        cls, info: ResolveInfo, instance: models.Promotion, data: dict, **kwargs
+    ):
+        cleaned_input = super().clean_input(info, instance, data, **kwargs)
+
+        errors: DefaultDict[str, List[ValidationError]] = defaultdict(list)
+        start_date = cleaned_input.get("start_date")
+        end_date = cleaned_input.get("end_date")
+        try:
+            validate_end_is_after_start(start_date, end_date)
+        except ValidationError as error:
+            error.code = PromotionCreateErrorCode.INVALID.value
+            errors["end_date"].append(error)
+
+        if rules := cleaned_input.get("rules"):
+            cleaned_rules, errors = cls.clean_rules(info, rules, errors)
+            cleaned_input["rules"] = cleaned_rules
+
+        if errors:
+            raise ValidationError(errors)
+
+        return cleaned_input
+
+    @classmethod
+    def clean_rules(
+        cls,
+        info: ResolveInfo,
+        rules_data: dict,
+        errors: DefaultDict[str, List[ValidationError]],
+    ):
+        cleaned_rules = []
+        for index, rule_data in enumerate(rules_data):
+            if channel_ids := rule_data.get("channels"):
+                channels = cls.clean_channels(info, channel_ids, index, errors)
+                rule_data["channels"] = channels
+
+            if "catalogue_predicate" not in rule_data:
+                errors["catalogue_predicate"].append(
+                    ValidationError(
+                        "The cataloguePredicate field is required.",
+                        code=PromotionCreateErrorCode.REQUIRED.value,
+                        params={"index": index},
+                    )
+                )
+            else:
+                if "reward_value_type" not in rule_data:
+                    errors["reward_value_type"].append(
+                        ValidationError(
+                            "The rewardValueType is required for when "
+                            "cataloguePredicate is provided.",
+                            code=PromotionCreateErrorCode.REQUIRED.value,
+                            params={"index": index},
+                        )
+                    )
+                if "reward_value" not in rule_data:
+                    errors["reward_value"].append(
+                        ValidationError(
+                            "The rewardValue is required for when cataloguePredicate "
+                            "is provided.",
+                            code=PromotionCreateErrorCode.REQUIRED.value,
+                            params={"index": index},
+                        )
+                    )
```

Comment:
Don't get this part. Why can't we set `catalogue_predicate`, `reward_value_type` and `reward_value` as required input fields?